api/v1alpha1/api.go

```diff
@@ -47,18 +47,68 @@ type LLMRouteSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self.schema == 'OpenAI'"
 	APISchema LLMAPISchema `json:"inputSchema"`
-	// HTTPRoute is the base HTTPRouteSpec (https://gateway-api.sigs.k8s.io/api-types/httproute/) in
-	// the Gateway API on which this LLMRoute will be implemented. AI Gateway controller will generate a HTTPRoute based
-	// on the configuration given here with the additional modifications to achieve the necessary jobs,
-	// notably inserting the AI Gateway external processor filter.
+	// Rules is the list of LLMRouteRule that this LLMRoute will match the traffic to.
+	// Each rule is a subset of the HTTPRoute in the Gateway API (https://gateway-api.sigs.k8s.io/api-types/httproute/).
 	//
-	// In the matching rules in the HTTPRoute here, `x-envoy-ai-gateway-llm-model` header
-	// can be used to describe the routing behavior.
+	// AI Gateway controller will generate a HTTPRoute based on the configuration given here with the additional
+	// modifications to achieve the necessary jobs, notably inserting the AI Gateway external processor filter.
 	//
-	// Currently, only the exact header matching is supported, otherwise the configuration will be rejected.
+	// In the matching conditions in the LLMRouteRule, `x-envoy-ai-gateway-llm-model` header
```

Comment:
we also match by `x-envoy-ai-gateway-llm-backend` and `x-envoy-ai-gateway-llm-model` is used for a more specific model level match. When user add these rules, they need to put more specific rules on the top right?

Reply:
well who sets `x-envoy-ai-gateway-llm-backend` in that case? Users are responsible setting the header and then they can do whatever they want not limited to the specific key `x-envoy-ai-gateway-llm-backend`?

Reply:
ye the key name does not matter, just to clarify the priority when there are multiple matches as we need to implement this in the extproc.

Reply:
https://github.com/envoyproxy/ai-gateway/pull/47/commits/d7bf9126ba0e6901679048e8f932f750a1ce7100 looks good?